auth/src/main/java/com/firebase/ui/auth/ui/idp/AuthMethodPickerActivity.java

```diff
@@ -77,7 +83,46 @@ protected void onCreate(Bundle savedInstanceState) {
         mSaveSmartLock = mActivityHelper.getSaveSmartLockInstance();
         findViewById(R.id.email_provider).setOnClickListener(this);
 
-        populateIdpList(mActivityHelper.getFlowParams().providerInfo);
+        Map<String, IdpConfig> providerIdToConfig = new HashMap<>();
+        List<IdpConfig> allConfiguredProviders = mActivityHelper.getFlowParams().providerInfo;
+        for (IdpConfig providerConfig : allConfiguredProviders) {
+            providerIdToConfig.put(providerConfig.getProviderId(), providerConfig);
+        }
+        List<IdpConfig> visibleProviders = new ArrayList<>();
+        if (mActivityHelper.getFlowParams().isReauth) {
+            // display the reauthReason if available
+            String reauthReason = mActivityHelper.getFlowParams().reauthReason;
+            if (!TextUtils.isEmpty(reauthReason)) {
+                TextView reauthReasonTextView = (TextView) findViewById(R.id.reauth_reason);
+                reauthReasonTextView.setText(reauthReason);
+                reauthReasonTextView.setVisibility(View.VISIBLE);
+            }
+
+            // For reauth flow we only want to show the IDPs which the user has associated with
+            // their account.
+            List<String> providerIds = mActivityHelper.getCurrentUser().getProviders();
+            if (providerIds.size() == 0) {
+                // zero providers indicates that it is an email account
+                visibleProviders.add(new IdpConfig.Builder(AuthUI.EMAIL_PROVIDER).build());
+            } else {
+                for (String providerId : providerIds) {
+                    IdpConfig idpConfig = providerIdToConfig.get(providerId);
+                    if (idpConfig == null) {
+                        Log.e(TAG, "User has provider " + providerId + " associated with their "
+                            + "account, but only the following IDPs have been configured: "
+                            + TextUtils.join(", ", providerIdToConfig.keySet()));
+                    } else {
+                        visibleProviders.add(idpConfig);
+                    }
+                }
+            }
+        } else {
+            visibleProviders = allConfiguredProviders;
+        }
+        if (visibleProviders.size() == 0) {
+            Log.e(TAG, "There are no configured providers associated with the user's account");
+        }
+        populateIdpList(visibleProviders);
```

Comment:
Before this happens I think we should check to see if `visibleProviders.size() == 1` and just launch the sign in right away if that is the case.